internet-shop/src/main/java/com/andersen/shop/dao/BasketDao.java

```diff
@@ -27,14 +38,39 @@ public BasketDao(ProductDAO productDAO, DataSource dataSource) {
         return products;
     }
 
-    public int addToBasket(long basketId, long productId) {
+    public int addToBasket(String username, long productId) {
         String sql = "INSERT INTO products_in_basket (basket_id, product_id, quantity) VALUES (?, ?, ?)";
-        return this.getJdbcTemplate().update(sql, basketId, productId, 0);
+        long basketId = 0;
+        try {
+            basketId = userDao.getUserID(username);
+        } catch (UserNotFoundException e) {
+            e.printStackTrace();
+        }
+        if (!checkIfProductIsInTheBasket(basketId, productId)) {
+            return this.getJdbcTemplate().update(sql, basketId, productId, 0);
+        }
+        return 0;
     }
 
-    public int deleteFromBasket(long basketId, long productId) {
+    public int deleteFromBasket(String username, long productId) {
         String sql = "DELETE FROM products_in_basket WHERE basket_id = ? and product_id = ?";
+        long basketId = 0;
+        try {
+            basketId = userDao.getUserID(username);
+        } catch (UserNotFoundException e) {
+            e.printStackTrace();
+        }
         return this.getJdbcTemplate().update(sql, basketId, productId);
     }
 
+    public boolean checkIfProductIsInTheBasket(long basketId, long productId) {
+        String sql = "SELECT pib.product_id FROM products_in_basket pib WHERE pib.basket_id = ?";
+        List<Long> list = this.getJdbcTemplate().query(sql, new Object[]{basketId}, (resultSet, i) -> resultSet.getLong(1));
+        for (Long id : list) {
+            if (id == productId) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
hm . is this part worth?
maybe to simplify it? e.g. return list.contatins(productId);